File: PrimaBank-Statistics/src/main/java/com/example/primabankstatistics/dto/ProductDTO.java
package com.example.primabankstatistics.dto;

import java.util.Objects;

public class ProductDTO {

    private String id;
    private String name;
    private String description;
    private double price;
    private String mode;

    public ProductDTO() {
    }

    public ProductDTO(String id, String name, String description, double price, String mode) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.price = price;
        this.mode = mode;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public String getMode() {
        return mode;
    }

    public void setMode(String mode) {
        this.mode = mode;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ProductDTO that = (ProductDTO) o;
        return Double.compare(that.price, price) == 0 && Objects.equals(id, that.id) && Objects.equals(name, that.name) && Objects.equals(description, that.description) && Objects.equals(mode, that.mode);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, name, description, price, mode);
    }

    @Override
    public String toString() {
        return "ProductDTO{" +
                "id='" + id + '\'' +
                ", name='" + name + '\'' +
                ", description='" + description + '\'' +
                ", price=" + price +
                ", mode='" + mode + '\'' +
                '}';
    }
}

File: PrimaBank-Marketing-Decision-Tree/app/dto/productDto.py
class ProductDTO:
    def __init__(self, id, name, description, price, mode):
        self.id = id
        self.name = name
        self.description = description
        self.price = price
        self.mode = mode

    def to_dict(self):
        return {
            'id': self.id,
            'name': self.name,
            'description': self.description,
            'price': self.price,
            'mode': self.mode
        }

    def get_id(self):
        return self.id

    def set_id(self, id):
        self.id = id

    def get_name(self):
        return self.name

    def set_name(self, name):
        self.name = name

    def get_description(self):
        return self.description

    def set_description(self, description):
        self.description = description

    def get_price(self):
        return self.price

    def set_price(self, price):
        self.price = price

    def get_mode(self):
        return self.mode

    def set_mode(self, mode):
        self.mode = mode

    def __eq__(self, other):
        if self is other:
            return True
        if not isinstance(other, ProductDTO):
            return False
        return (
            self.price == other.price
            and self.id == other.id
            and self.name == other.name
            and self.description == other.description
            and self.mode == other.mode
        )

    def __hash__(self):
        return hash((self.id, self.name, self.description, self.price, self.mode))

    def __str__(self):
        return f"ProductDTO{{id='{self.id}', name='{self.name}', description='{self.description}', price={self.price}, mode='{self.mode}'}}"
